nxcomp/src/ChannelEndPoint.cpp

```diff
@@ -140,7 +140,10 @@ ChannelEndPoint::getSpec(char **socketUri) const {
       snprintf(newSocketUri, length+1, "tcp:%s:%ld", hostName, port);
 
     if (socketUri)
+    {
+      SAFE_FREE(*socketUri);
```

Comment:
Uh, this might be dangerous.

I'd drop this commit, since it's better to let callers handle their memory and make sure that `socketUri` is free'd in the callers.